utils/python/wait_for_tests.py

```diff
@@ -429,6 +429,12 @@ def wait_for_tests_impl(test_paths, no_wait=False, check_throughput=False, check
 
     return test_results
 
+def update_testspec_xml(test_results):
```

Comment:
this is incomplete code - i guess I got distracted and forgot about it.  Maybe I'll pull the testspec changes out of this PR and do it separately.